velox/serializers/PrestoSerializer.cpp

```diff
@@ -625,6 +628,65 @@ void read<StringView>(
   }
 }
 
+template <>
+void read<OpaqueType>(
+    ByteInputStream* source,
+    const TypePtr& type,
+    vector_size_t resultOffset,
+    const uint64_t* incomingNulls,
+    int32_t numIncomingNulls,
+    memory::MemoryPool* pool,
+    const SerdeOpts&,
+    VectorPtr& result) {
+  // Opaque values are serialized by first converting them to string
+  // them serializing them as if they were string. The deserializable
+  // does the reverse operation.
+
+  const int32_t size = source->read<int32_t>();
+  const int32_t numNewValues = sizeWithIncomingNulls(size, numIncomingNulls);
+
+  result->resize(resultOffset + numNewValues);
+
+  auto opaqueType = std::dynamic_pointer_cast<const OpaqueType>(type);
+  auto deserialization = opaqueType->getDeserializeFunc();
+
+  auto flatResult = result->as<FlatVector<std::shared_ptr<void>>>();
+  BufferPtr values = flatResult->mutableValues(resultOffset + size);
+
+  auto rawValues = values->asMutable<std::shared_ptr<void>>();
+  std::vector<int32_t> offsets;
+  int32_t lastOffset = 0;
+  for (int32_t i = 0; i < numNewValues; ++i) {
+    // Set the first int32_t of each StringView to be the offset.
+    if (incomingNulls && bits::isBitNull(incomingNulls, i)) {
+      offsets.push_back(lastOffset);
+      continue;
+    }
+    lastOffset = source->read<int32_t>();
+    offsets.push_back(lastOffset);
+  }
+  readNulls(
+      source, size, resultOffset, incomingNulls, numIncomingNulls, *flatResult);
+
+  const int32_t dataSize = source->read<int32_t>();
+  if (dataSize == 0) {
+    return;
+  }
+
+  const int64_t rawStringSizeBytes = dataSize * sizeof(char);
+  char* rawString = (char*)pool->allocate(rawStringSizeBytes);
```

Comment:
nit: instead of directly using the pool's allocate API that returns a raw pointer, I would suggest create a buffer and using a BufferPtr instead since its much more safer as the space will be freed when BufferPtr is destroyed, otherwise if any of the steps throw after this then it can introduce a memory leak.

Its like using a shared_ptr vs a raw ptr and has similar benefits.

You can use:
```
BufferPtr newBuffer = AlignedBuffer::allocate<char>(dataSize, pool());
```